internal/core/data/controller/http/event.go

```diff
@@ -71,8 +73,15 @@ func (ec *EventController) AddEvent(w http.ResponseWriter, r *http.Request) {
 	sourceName := vars[common.SourceName]
 
 	var addEventReqDTO requestDTO.AddEventRequest
+	var err errors.EdgeX
+
+	dataBytes, readErr := io.ReadAll(r.Body)
+	if readErr != nil {
+		err = errors.NewCommonEdgeX(errors.KindIOError, "AddEventRequest I/O reading failed", nil)
+	} else {
+		err = utils.CheckPayloadSize(dataBytes, config.MaxEventSize*1000)
+	}
```

Comment:
r.ContentLength is a better option, but why do we limit the check for CBOR only?

Reply:
@lenny-intel Although the request usually contains ContentLength, it's not a required field.
Should we depend on it?  It might be unknown.
https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.4

> 	// ContentLength records the length of the associated content.
> 	// The value -1 indicates that the length is unknown.
> 	// Values >= 0 indicate that the given number of bytes may
> 	// be read from Body.
> 	//
> 	// For client requests, a value of 0 with a non-nil Body is
> 	// also treated as unknown.
> 	ContentLength [int64](https://pkg.go.dev/builtin#int64)

I suggest to check ContentLength first, and keep the same `utils.CheckPayloadSize(dataBytes, config.MaxEventSize*1000)` step

Reply:
Yes, check first, if -1 use `utils.CheckPayloadSize`